### PR TITLE
[JAX] Adding Gated/Non-gated ReLU, Quick GeLU, Squared ReLU

### PR DESCRIPTION
**Pull request by phu0ngng**
# Description

This PR adds Gated and Non-gated ReLU, Quick GeLU, Squared ReLU.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Changes

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by denera — APPROVED** [review]
LGTM, pending the refactoring for `NVTE_Activation_Enum`.

Please run the full TE CI for all frameworks to make sure TE/common changes in this PR don't break anything outside if JAX.

File: transformer_engine/jax/csrc/modules.h [hunk 1]
```diff
@@ -43,26 +43,44 @@ struct Shape {
     }
 };
 
+enum class NVTE_Activation_Enum {
```
[the remark this answers is not in the document]
Comment (phu0ngng) [reply transformer_engine/jax/csrc/modules.h]: Hi. Thanks. I implemented your suggestion.

**Pull request merged**